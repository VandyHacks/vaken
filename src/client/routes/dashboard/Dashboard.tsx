import React, { FunctionComponent } from 'react';
import { Switch, Route } from 'react-router-dom';
import { HackerDash } from './HackerDash';
import { OrganizerDash } from './OrganizerDash';

export const Dashboard: FunctionComponent = (): JSX.Element => {
	return (
		<>
			<Switch>
				<Route path="/dashboard/organizer" component={OrganizerDash} />
				<Route path="/dashboard" component={HackerDash} />
			</Switch>
		</>
	);
};

export default Dashboard;
