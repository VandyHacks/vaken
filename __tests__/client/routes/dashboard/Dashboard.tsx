import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Dashboard from '../../../../src/client/routes/dashboard/Dashboard';

describe('Test Dashboard', () => {
	it('renders hackerdash on /dashboard route', async () => {
		// jest.mock('react-virtualized/dist/commonjs/AutoSizer', () => ({
		// 	render: () => <div />,
		// }));
		const component = renderer
			.create(
				<MemoryRouter initialEntries={['/dashboard']}>
					<Dashboard />
				</MemoryRouter>
			)
			.toJSON();

		expect(component).toMatchSnapshot();
	});
});
