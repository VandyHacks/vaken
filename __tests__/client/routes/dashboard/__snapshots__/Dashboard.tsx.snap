// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Dashboard renders hackerdash on /dashboard route 1`] = `
<div
  className="sc-bdVaJa sc-bwzfXH sc-ifAKCX kHHPcv"
>
  <div
    className="sc-bdVaJa sc-bwzfXH sc-ifAKCX sc-jzJRlG knwSVK"
    height="min-content"
    width="36rem"
  >
    <div
      className="sc-bdVaJa sc-bwzfXH dgnCwc"
    >
      <h1
        className="sc-cSHVUG fiKfBA"
        fontSize="1.8rem"
      >
        Your application status is:
      </h1>
      <div
        className="sc-VigVT sc-jTzLTM bHJZvB"
      >
        <div
          className="sc-htoDjs ewDcVm"
          color="#FF647C"
          fontSize="1.8rem"
          fontWeight="bold"
        >
          INCOMPLETE
        </div>
      </div>
      <img
        alt="Man wearing hoodie at holographic computer"
        src={[Function]}
      />
      <h1
        className="sc-kAzzGY jWKBgv"
        color="#3F3356"
        fontSize="1rem"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          You still need to finish your application.
        </span>
        <br />
        The deadline is January 1, 2020 at 11:59 pm (EST).
      </h1>
      <a
        href="/application"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <button
          className="sc-gqjmRU sc-fjdhpX hFJOLF"
          color="white"
          fontSize="1.4em"
        >
          <div
            className="sc-htoDjs kSfDbb"
            color="white"
            fontSize="1.4em"
          >
            Complete your application
          </div>
        </button>
      </a>
    </div>
  </div>
</div>
`;
